File: backend/app/schemas/human_review.py
from pydantic import BaseModel, ConfigDict
from uuid import UUID
from typing import Optional, Any, Dict
from datetime import datetime

class HumanReviewCreate(BaseModel):
    topic: str
    user_question: str
    priority: str = "Normal"  # Normal, Urgent
    assessment_id: Optional[UUID] = None
    ai_assessment: Optional[Dict[str, Any]] = None

class HumanReviewUpdate(BaseModel):
    status: Optional[str] = None  # new, assigned, in_review, completed, needs_info
    facilitator_notes: Optional[str] = None
    final_guidance: Optional[str] = None
    priority: Optional[str] = None
    facilitator_id: Optional[UUID] = None

class HumanReviewResponse(BaseModel):
    id: UUID
    assessment_id: Optional[UUID] = None
    user_id: UUID
    facilitator_id: Optional[UUID] = None
    status: str
    topic: Optional[str] = None
    user_question: str
    ai_assessment: Optional[Dict[str, Any]] = None
    facilitator_notes: Optional[str] = None
    final_guidance: Optional[str] = None
    priority: str
    assigned_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/humanReview.ts
import { apiClient } from './client';

export interface HumanReviewItem {
  id: string;
  assessment_id?: string;
  user_id: string;
  facilitator_id?: string;
  status: 'new' | 'assigned' | 'in_review' | 'completed' | 'needs_info';
  topic?: string;
  user_question: string;
  ai_assessment?: Record<string, any>;
  facilitator_notes?: string;
  final_guidance?: string;
  priority: 'normal' | 'urgent' | string;
  assigned_at?: string;
  completed_at?: string;
  created_at: string;
}

export interface HumanReviewCreateRequest {
  topic: string;
  user_question: string;
  priority?: string;
  assessment_id?: string;
  ai_assessment?: Record<string, any>;
}

export interface HumanReviewUpdateRequest {
  status?: string;
  facilitator_notes?: string;
  final_guidance?: string;
  priority?: string;
  facilitator_id?: string;
}

export const humanReviewApi = {
  getReviews: async (status?: string): Promise<HumanReviewItem[]> => {
    const params = status ? { status } : {};
    const response = await apiClient.get<HumanReviewItem[]>('/human-review', { params });
    return response.data;
  },
  getReview: async (id: string): Promise<HumanReviewItem> => {
    const response = await apiClient.get<HumanReviewItem>(`/human-review/${id}`);
    return response.data;
  },
  createReview: async (data: HumanReviewCreateRequest): Promise<HumanReviewItem> => {
    const response = await apiClient.post<HumanReviewItem>('/human-review', data);
    return response.data;
  },
  updateReview: async (id: string, data: HumanReviewUpdateRequest): Promise<HumanReviewItem> => {
    const response = await apiClient.patch<HumanReviewItem>(`/human-review/${id}`, data);
    return response.data;
  },
};
